staging/src/k8s.io/client-go/transport/round_trippers.go

```diff
@@ -61,9 +65,27 @@ func HTTPWrappersForConfig(config *Config, rt http.RoundTripper) (http.RoundTrip
 		len(config.Impersonate.Extra) > 0 {
 		rt = NewImpersonatingRoundTripper(config.Impersonate, rt)
 	}
+	rt = NewOtelRoundTripper(rt)
 	return rt, nil
 }
 
+type otelWrapper struct {
+	otelRT http.RoundTripper
+	baseRT http.RoundTripper
+}
+
+// NewOtelRoundTripper returns an otelhttp-wrapped round tripper that implements
+// WrappedRoundTripper
+func NewOtelRoundTripper(rt http.RoundTripper) http.RoundTripper {
```

Comment:
note to reviewers: This is required because unit tests "unwrap" the RoundTripper to inspect the underlying http Transport